src/Snak/ExtendedPropertyValueSnak.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Wikibase\DataModel\Snak;
+
+use DataValues\DataValue;
+use InvalidArgumentException;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * @since 3.1
```

Comment:
Please add a description.

For example, I see that this class does not override `getType`, therefore will be of type "value" and not distinguishable from `PropertyValueSnak`. As far as I know this is intended but should be documented.